backend/dataall/modules/dataset_sharing/services/share_managers/s3_bucket_share_manager.py

```diff
@@ -72,95 +73,71 @@ def grant_s3_iam_access(self):
         logger.info(
             f'Grant target role {self.target_requester_IAMRoleName} access policy'
         )
-        existing_policy = IAM.get_role_policy(
+        bucket_policy_name = ConsumptionRole.generate_policy_name(self.target_environment.environmentUri,
+                                                                  self.target_requester_IAMRoleName)
+
+        logger.info(f'Share policy name is {bucket_policy_name}')
+        version_id, policy_document = IAM.get_managed_policy_default_version(
             self.target_account_id,
-            self.target_requester_IAMRoleName,
-            IAM_S3BUCKET_ROLE_POLICY,
-        )
+            bucket_policy_name)
+
         key_alias = f"alias/{self.target_bucket.KmsAlias}"
         kms_client = KmsClient(self.source_account_id, self.source_environment.region)
         kms_key_id = kms_client.get_key_id(key_alias)
 
-        if existing_policy:  # type dict
-            s3_target_resources = [
-                f"arn:aws:s3:::{self.bucket_name}",
-                f"arn:aws:s3:::{self.bucket_name}/*"
-            ]
+        s3_target_resources = [
+            f"arn:aws:s3:::{self.bucket_name}",
+            f"arn:aws:s3:::{self.bucket_name}/*"
+        ]
+
+        share_manager = ShareManagerUtils(
+            self.session,
+            self.dataset,
+            self.share,
+            self.source_environment,
+            self.target_environment,
+            self.source_env_group,
+            self.env_group
+        )
 
-            share_manager = ShareManagerUtils(
-                self.session,
-                self.dataset,
-                self.share,
-                self.source_environment,
-                self.target_environment,
-                self.source_env_group,
-                self.env_group
-            )
-            share_manager.add_missing_resources_to_policy_statement(
-                resource_type=self.bucket_name,
-                target_resources=s3_target_resources,
-                existing_policy_statement=existing_policy["Statement"][0],
-                iam_role_policy_name=IAM_S3BUCKET_ROLE_POLICY
-            )
+        share_manager.add_missing_resources_to_policy_statement(
+            resource_type=self.bucket_name,
+            target_resources=s3_target_resources,
+            existing_policy_statement=policy_document["Statement"][0],
+            iam_role_policy_name=IAM_S3BUCKET_ROLE_POLICY
```

Comment:
Similar to above comment with access points and `IAM_ACCESS_POINT_ROLE_POLICY`

Reply:
changed